src/ngraph/runtime/reference/topk.hpp

```diff
@@ -33,7 +33,18 @@ namespace ngraph
             template <typename T, typename U>
             static bool compare_max(const std::tuple<T, U>& a, const std::tuple<T, U>& b)
             {
-                return a > b;
+                if (std::get<0>(a) > std::get<0>(b))
```

Comment:
Tuple comparison is supposed to be lexical so I'm not sure why you should need all this, or even `compare_max`.

Reply:
As discussed on slack - will go back the solution with direct comparison of elements at index 0 and disable the clang warning using pragma.